File: bindings/java-ffm/src/main/java/org/maplibre/nativeffi/geo/Feature.java
package org.maplibre.nativeffi.geo;

import java.util.List;
import java.util.Objects;
import org.maplibre.nativeffi.json.JsonValue;

/** Immutable GeoJSON feature descriptor. */
public record Feature(
    Geometry geometry, List<JsonValue.Member> properties, FeatureIdentifier identifier) {
  public Feature {
    Objects.requireNonNull(geometry, "geometry");
    properties = List.copyOf(properties);
    identifier = identifier == null ? FeatureIdentifier.nullValue() : identifier;
  }
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/geo/Feature.kt
package org.maplibre.nativeffi.geo

import org.maplibre.nativeffi.json.JsonValue

/** Immutable GeoJSON feature descriptor. */
public data class Feature(
  public val geometry: Geometry,
  public val properties: List<JsonValue.Member>,
  public val identifier: FeatureIdentifier = FeatureIdentifier.Null,
)
